packages/peer-exchange/src/waku_peer_exchange_discovery.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  PeerDiscovery,
+  PeerDiscoveryEvents,
+  symbol,
+} from "@libp2p/interface-peer-discovery";
+import { PeerInfo } from "@libp2p/interface-peer-info";
+import { PeerProtocolsChangeData } from "@libp2p/interface-peer-store";
+import { EventEmitter } from "@libp2p/interfaces/events";
+import { PeerExchangeComponents } from "@waku/interfaces";
+import debug from "debug";
+
+import { PeerExchangeCodec } from "./waku_peer_exchange";
+
+const log = debug("waku:peer-exchange-discovery");
+
+interface Options {
+  /**
+   * Tag a bootstrap peer with this name before "discovering" it (default: 'bootstrap')
+   */
+  tagName?: string;
+
+  /**
+   * The bootstrap peer tag will have this value (default: 50)
+   */
+  tagValue?: number;
+
+  /**
+   * Cause the bootstrap peer tag to be removed after this number of ms (default: 2 minutes)
+   */
+  tagTTL?: number;
+}
+
+const DEFAULT_BOOTSTRAP_TAG_NAME = "peer-exchange";
+const DEFAULT_BOOTSTRAP_TAG_VALUE = 50;
+const DEFAULT_BOOTSTRAP_TAG_TTL = 120000;
+
+export class PeerExchangeDiscovery
+  extends EventEmitter<PeerDiscoveryEvents>
+  implements PeerDiscovery
+{
+  private readonly components: PeerExchangeComponents;
+  private readonly options: Options;
+  private isStarted: boolean;
+
+  private readonly eventHandler = async (
+    event: CustomEvent<PeerProtocolsChangeData>
+  ): Promise<void> => {
+    const { protocols } = event.detail;
+    if (!protocols.includes(PeerExchangeCodec)) return;
+
+    const { peerId } = event.detail;
```

Comment:
```suggestion
    const { protocols, peerId } = event.detail;
    if (!protocols.includes(PeerExchangeCodec)) return;
```
 
nipick: Just deconstruct all data from event upfront.